libs/common/keycloak.go

```diff
@@ -17,11 +18,21 @@ import (
 // To add a claim to this struct please add
 // a client scope (and mapping) to the client in keycloak first
 type AccessTokenClaims struct {
-	// TODO
+	// Sub is the user id
+	Sub        string `json:"sub"`
+	GivenName  string `json:"given_name"`
+	FamilyName string `json:"family_name"`
+	Email      string `json:"email"`
 }
 
 func (a AccessTokenClaims) AsExpected() error {
```

Comment:
Right now, you create a struct of `AccessTokenClaims`, set the values and run `.AsExpected()`. Do you think it makes sense to handle these kinds of validation inside a constructor? Something like `NewAccessTokenClaims` where you pass `Sub` and `Email`, test against your cases and return a new struct of `AccessTokenClaims`.

Reply:
`AccessTokenClaims` should only be constructed in [VerifyAccessToken](https://github.com/helpwave/services/blob/4c705e8e1d3c10755570719586da10d71a6c3c98/libs/common/keycloak.go#L75) and filled by the token library using a reference. So this is not possible.